R/bambu-processReads_utilityConstructReadClasses.R

```diff
@@ -268,21 +279,25 @@ getUnsplicedReadClassByReference <- function(granges, grangesReference,
     ## create single exon read class by using the minimum end
     ## and maximum start of all overlapping exons (identical to
     ## minimum equivalent class)
-    hitsDF <- hitsDF %>% select(queryHits, chr, start, end, strand) %>%
+    hitsDF <- hitsDF %>% dplyr::select(queryHits, chr, start, end, strand) %>%
         group_by(queryHits, chr, strand) %>%
         summarise(start = max(start), end = min(end), .groups = "drop") %>%
         group_by(chr, start, end, strand) %>%
         mutate(readClassId = paste0("rc", confidenceType, ".", 
             cur_group_id())) %>% ungroup()
     readIds <- mcols(granges[hitsDF$queryHits])$id
-    hitsDF <- hitsDF %>% select(chr, start, end, strand, readClassId) %>%
-        group_by(readClassId) %>% mutate(readCount = n()) %>% 
-        ungroup() %>% distinct() %>% 
+    hitsDF$alignmentStrand = as.factor(strand(granges))[hitsDF$queryHits]
+    #previously it took the first rows strand for a read class id which could be wrong
+    #coded in an alternative but its likely very slow....
+    hitsDF <- hitsDF %>% dplyr::select(chr, start, end, strand, 
+        readClassId, alignmentStrand) %>%
+        group_by(readClassId) %>% summarise(start = start[1], end = end[1], 
+        strand = names(which.max(table(strand))), chr = chr[1], readCount = n(),
```

Comment:
Resolved this in commit 4c1e49e
I did not change strand to NA because if you do it produces error messages saying strand cannot be anything other than +/-/* and coverts NA to * anyway.